sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOTranslation.java

```diff
@@ -274,10 +274,16 @@ public TypedRead<?> fromConfigRow(Row configRow, PipelineOptions options) {
         if (queryTempDataset != null) {
           builder = builder.setQueryTempDataset(queryTempDataset);
         }
-        String queryTempProject = configRow.getString("query_temp_project");
-        if (queryTempProject != null) {
-          builder = builder.setQueryTempProject(queryTempProject);
+
+        if (TransformUpgrader.compareVersions(updateCompatibilityBeamVersion, "2.57.0") >= 0) {
+          // This property was added for Beam 2.57.0 hence not available when
+          // upgrading the transform from previous Beam versions.
+          String queryTempProject = configRow.getString("query_temp_project");
+          if (queryTempProject != null) {
```

Comment:
Sounds like this null check is probably redundant, if the configRow crashes out? Or does the row have three states? (null, empty, string)

Reply:
Yeah, in this case the null check does not help since the field simply doesn't exist. I guess if the field has a null state it can still be good to have (unless we explicitly want to set null values).

Reply:
Hmm, when rows expand the nullable fields should be filled with nulls. I know we have some support for this in the data plane in Dataflow. Perhaps there is something different here because it is not data plane?

Reply:
What do you mean by "rows expand" here ? Creating a POJO from a "from Row" function created using the `SchemaRegistry.fromRowFunction` [util](https://github.com/apache/beam/blob/911c525348b81afd48a414018ba579c3b78e3262/sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/SchemaRegistry.java#L269) ?

Note we are trying to get specific fields from the `Row` which map to construction fields here. We don't have a single construction object to directly parse the Row into. So we should probably just add utils to handle schema-errors due to missing fields here.